x-pack/plugins/monitoring_collection/server/routes/dynamic_route.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { Observable } from 'rxjs';
+import { JsonObject } from '@kbn/utility-types';
+import { schema } from '@kbn/config-schema';
+import { IRouter, ServiceStatus } from '../../../../../src/core/server';
+import { getESClusterUuid, getKibanaStats } from '../lib';
+import { MetricResult } from '../plugin';
+
+export function registerDynamicRoute({
+  router,
+  config,
+  overallStatus$,
+  getMetric,
+}: {
+  router: IRouter;
+  config: {
+    allowAnonymous: boolean;
+    kibanaIndex: string;
+    kibanaVersion: string;
+    uuid: string;
+    server: {
+      name: string;
+      hostname: string;
+      port: number;
+    };
+  };
+  overallStatus$: Observable<ServiceStatus>;
+  getMetric: (
+    type: string
+  ) => Promise<Array<MetricResult<JsonObject>> | MetricResult<JsonObject> | undefined>;
+}) {
+  router.get(
+    {
+      path: `/api/monitoring_collection/{type}`,
+      options: {
```

Comment:
@jportner @legrego We're adding a new HTTP endpoint that will expose _somewhat_ arbitrary metrics about various systems within Kibana. Currently, we're only slated to only add metrics for the alerting framework ([exact details here](https://github.com/elastic/kibana/issues/122362#issuecomment-1020359743)) and none of these metrics expose any user-specific data (like named things). The intended consumer of this API will be Metricbeat, as part of the Stack Monitoring Metricbeat monitoring story. The other endpoint Metricbeat talks to is the `/api/stats` which exists in [the usage collection](https://github.com/elastic/kibana/blob/main/src/plugins/usage_collection/server/routes/stats/stats.ts#L73) plugin. Should we copy the same security settings for this new route? Should we be concerned about RBAC now or in the future for this route?

Reply:
> Should we copy the same security settings for this new route?

I would suggest not following the same pattern as `/api/stats`, because I don't expect that Metricbeat will need anonymous access (metricbeat should always authenticate when security is enabled, I presume).

What privileges does Metricbeat have when properly configured today?

> Should we be concerned about RBAC now or in the future for this route?

If it's shipping in a release, then we can't defer RBAC.

Reply:
> What privileges does Metricbeat have when properly configured today?

Metricbeat requires configuration to determine access - the requests are made using the configured username/password.

From https://www.elastic.co/guide/en/kibana/current/monitoring-metricbeat.html:

> If the Elastic security features are enabled, you must also provide a user ID and password so that Metricbeat can collect metrics successfully:

> Create a user on the production cluster that has the remote_monitoring_collector [built-in role](https://www.elastic.co/guide/en/elasticsearch/reference/8.0/built-in-roles.html). Alternatively, use the remote_monitoring_user [built-in user](https://www.elastic.co/guide/en/elasticsearch/reference/8.0/built-in-users.html).
Add the username and password settings to the Kibana module configuration file.

From a Metricbeat perspective, both `/api/stats` and this endpoint should behave the same so I'm not sure I'm following your comment `I would suggest not following the same pattern as /api/stats`, unless the `/api/stats` is incorrectly configured. 

> If it's shipping in a release, then we can't defer RBAC.

Perhaps I'm asking the wrong question here, because the particular data returned from this API doesn't contain anything that is user-centric (that would require a `consumer`/`producer` field), but this _might_ change in the future and I don't know how to handle that, cc @kobelb @XavierM who might be able to answer too

Reply:
Thanks for the context @chrisronline. 

> From a Metricbeat perspective, both /api/stats and this endpoint should behave the same so I'm not sure I'm following your comment I would suggest not following the same pattern as /api/stats, unless the /api/stats is incorrectly configured.

From a Metricbeat perspective, I think they will, because our docs state that we require Metricbeat to authenticate to Kibana if security features are enabled. The `/api/stats` endpoint can optionally be called without any authentication, and that's the part I didn't want to copy over here. I think this "feature" is primarily used by load balancers and other non-Elastic systems that need to ping Kibana instances to determine readiness/alive-ness.

> Perhaps I'm asking the wrong question here, because the particular data returned from this API doesn't contain anything that is user-centric (that would require a consumer/producer field), but this might change in the future and I don't know how to handle that

Good to know that nothing user-centric is returned here. We are returning Kibana-centric information though, so I feel that there is some level of authorization necessary here. What authorization should actually be required is a good question. @gmmorris / @mikecote what privileges are required for alerting's health check today, can you remind me? I know we want to change it eventually. Do we just require any set of valid credentials?

Reply:
The [alerting health endpoint](https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/routes/health.ts#L55) doesn't have explicit permissions but does require the authenticated user has access to at least one rule type

The [task manager health endpoint](https://github.com/elastic/kibana/blob/main/x-pack/plugins/task_manager/server/routes/health.ts#L117) doesn't seem to have any access restrictions.

FWIW, in the initial release of this feature, the data returned from this API will consist of counts from querying the task manager index, and also new counters (added in a subsequent PR) that tally up rule execution successes and rule execution failures

Reply:
Thanks for being patient with me @chrisronline, and for finding those references for me. I'd like to hear from Joe when he has a chance to weigh in, but it feels reasonable to me to follow what the task manager health endpoint is doing, for now: require authentication, but no specific authorization. I don't think we have a great mechanism to grant this role a special privilege for this API today, without also permitting it to login to Kibana, which I don't think we want.

If we add more interesting data here, then it'd probably be worthwhile to revisit this.

Reply:
>What authorization should actually be required is a good question. @gmmorris / @mikecote what privileges are required for alerting's health check today, can you remind me? I know we want to change it eventually. Do we just require any set of valid credentials?

@legrego, @chrisronline confirmed the behaviour that we currently have. For further restricting, we were tracking it under https://github.com/elastic/kibana/issues/102100 and opted to track telemetry for now until we understand how much an impact further restrictions would have on our users (https://github.com/elastic/kibana/pull/119349).

Reply:
> I'd like to hear from Joe when he has a chance to weigh in, but it feels reasonable to me to follow what the task manager health endpoint is doing, for now: require authentication, but no specific authorization.

Yeah, I agree.